packages/toggle/src/types.ts

```diff
@@ -0,0 +1,56 @@
+import { Either, HTMLElementProps } from '@leafygreen-ui/lib';
+
+export const Size = {
+  Default: 'default',
+  Small: 'small',
+  XSmall: 'xsmall',
+} as const;
+
+export type Size = typeof Size[keyof typeof Size];
+
+interface BaseToggleProps {
+  /**
+   * Sets the size of the toggle.
+   *
+   * default: `'default'`
+   */
+  size?: Size;
+
+  /**
+   * Determines if the Toggle will render the dark mode styles.
+   *
+   * default: `false`
+   */
+  darkMode?: boolean;
+
+  /**
+   * Sets the checked state of the Toggle.
+   */
+  checked?: boolean;
+
+  /**
+   * Disables the Toggle.
+   *
+   * default: `false`
+   */
+  disabled?: boolean;
+
+  /**
+   * `onChange` fires when the internally-managed `checked` state of the component is updated. Receives the updated checked state of the toggle as its first argument, and the associated mouse event as the second.
+   */
+  onChange?: (
+    checked: boolean,
+    mouseEvent: React.MouseEvent<HTMLButtonElement>,
+  ) => void;
+
+  /**
+   * Adds a className to the outermost element.
+   */
+  className?: string;
+}
+
+export type ToggleProps = Either<
+  BaseToggleProps &
+    Omit<HTMLElementProps<'button', never>, keyof BaseToggleProps>,
+  'aria-label' | 'aria-labelledby'
+>;
```

Comment:
I believe we need to do this because `button` clashes with the `onChange` prop. I could do this:
```interface BaseToggleProps extends Omit<HTMLElementProps<'button', never>, 'onChange'>``` but then we have to be explicit with what to omit. I think we should leave it the way it is.